core/includes/update.inc

```diff
@@ -294,6 +294,88 @@ function update_module_add_to_system($modules = array()) {
   }
 }
 
+/** 
+ * Checks for disabled dependencies during a Drupal 7 upgrade.
+ * 
+ * @return string
+ *    A themed message that lists modules that will be enabled,
+ *    to be displayed in the 'info' page of the update process. 
+ */
+
+function update_upgrade_check_dependencies() {
+  $status_report = '';
+  if (backdrop_get_installed_schema_version('system') > 7000) {
+    $modules_to_enable = update_upgrade_modules_to_enable();
+    
+    if (empty($modules_to_enable)) {
+      return '';
+    }
+
+    $files = system_rebuild_module_data();
+    $module_list = '';
+    foreach ($modules_to_enable as $module_to_enable => $dependents) {
+      $module_list .= '<strong>' . $files[$module_to_enable]->info['name'] . '. ' . t('Required by:') . '</strong>';
+      $module_list .= theme('item_list', array('items' => $dependents));
+    }
+    $status_report = '<h4>' . t('Some modules need to be enabled') . '</h4>';
+    $status_report .= t('The following Backdrop modules will be enabled in order for the site to be upgraded successfully:') . '</p>';
+    $status_report .= $module_list;
+  }
+  return $status_report;
+}
+
+/**
+ * Enables needed dependencies.
+ */
+function update_upgrade_enable_dependencies() {
+  if (backdrop_get_installed_schema_version('system') > 7000 && !state_get('update_upgrade_enable_dependencies', FALSE)) {
+    // When upgrading a D7 site we need to enable entity.module at this point
+    // since the calls to 'module_enable()' below need some of entity's functions.
+    // In a non-upgrade situation, entity.module will be enabled later,
+    // inside the hook implementation system_update_1000(). 
+    module_enable(array('entity'));
```

Comment:
Seems like this use `update_module_enable()` like `system_update_1000()` does, which skips `hook_module_enable()` hooks.

Reply:
We had some discussion of this issue at https://github.com/backdrop/backdrop-issues/issues/5499#issuecomment-1052769082 and following comments. Looked like using `update_module_enable()` was going to be more complicated. Would you review those comments and comment further?

Reply:
The issue with using `update_module_enable` to enable `entity` at this point is that this function will actually not load the code of the `entity` module - it only changes the `system` table in the database. So you still will get an `Call to undefined function entity_info_cache_clear()` when this process tries to enable Backdrop modules required by the Standard installation such as Redirect and others (those modules are not enabled by the current update.php code  - see the discussion in the issue associated to this PR).

So, the only way to use `update_module_enable(array('entity');` is to manually load the code with `backdrop_load('module', 'entity');` right after. I've tested this and it seems to work fine (and it avoids firing the enable hooks). 

Still, I'm not sure if this is preferable. Any thoughts?

Reply:
By the way, `module_enable('entity')` is needed here due to the new code below, which enables Backdrop's core modules required by Standard installation.  

Ironically, enabling `entity` here is  ONLY needed if your D7 site did not use Entity API. If your D7 used Entity API (that is, D7's `entity` contrib module), the D7 database will already have `entity` as enabled, and the Backdrop `entity` code will be loaded right away at the start of  the update.php code.